SpatialGDK/Source/SpatialGDK/Public/EngineClasses/AbstractPackageMapClient.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "SpatialConstants.h"
+#include "WorkerSDK/improbable/c_worker.h"
+
+#include "Engine/PackageMapClient.h"
+#include "UObject/Object.h"
+
+#include "AbstractPackageMapClient.generated.h"
+
+UCLASS()
+class SPATIALGDK_API UAbstractPackageMapClient : public UPackageMapClient
```

Comment:
The resulting name is a bit unusual, since UAbstractPackageMapClient is not really an abstract class.

Reply:
It _should_ be in the sense that the function definitions here _should_ be pure virtual (though I haven't got them like that currently). Happy to name changed either way - suggestions?

Reply:
I hoped you'd have some suggestions :) I'll let you know if I come up with any.

Reply:
the goal is for this to be a provider interface to the EnityIdMap, the fact that it's implemented in the PackageMapClient is convenience. So maybe EntityIdMappingProvider?  (Or maybe Oracle, but that may be a naming world we don't want to go to.)

Reply:
I'd prefer the Abstract prefix - I've updated the classes to be `abstract` and the remaining suffix makes the relative hierarchies clearer imo

Reply:
SGTM.

Reply:
I think the name is wrong for a different reason: this isn't an abstract package map (generally `AbstractFoo` deriving from `Foo` seems like a bad thing). This is just one function from the interface of a class that derives from `PackageMapClient`. that we are trying to hide from something else. You could maybe call it an abstract spatial package map or something because it inherits `PackageMapClient` but I'm not sure there is a good name for this as it feels like we are hacking something strange together to cover up other issues.